### PR TITLE
[NFC]: Cleanup RewriteTensorPointer

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 1]
```diff
@@ -425,54 +426,59 @@ class TritonIntelGPURewriteTensorPointerPass
...
+
+  Operation *rewriteOp(OpBuilder &builder, tt::StoreOp op,
+                       std::stack<Operation *> &eraser) {
+    if (!valueToRemove.count(op->getOperand(0)))
+      return nullptr;
+
+    // Get info from previous results
+    auto ptr = op->getOperand(0);
+    assert(rewritedInfo.count(ptr) &&
+           "Expecting LoadOp/StoreOp ptr in rewritedInfo");
```
Comment (LiyangLingIntel) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:459]:
```suggestion
           "Expecting LoadOp ptr in rewritedInfo");
```

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 2]
```diff
@@ -425,54 +426,59 @@ class TritonIntelGPURewriteTensorPointerPass
            "Expecting LoadOp/StoreOp ptr in rewritedInfo");
```
Comment (LiyangLingIntel) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:426]:
```suggestion
           "Expecting LoadOp ptr in rewritedInfo");
```

**Review by LiyangLingIntel — APPROVED** [review]
LGTM. Only some tiny changes as comments. Thanks for refactoring.

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 3]
```diff
@@ -666,36 +672,36 @@ class TritonIntelGPURewriteTensorPointerPass
...
-      return rewriteLoadStoreOp(builder, op, eraser);
-    } else if (op->getDialect()->getNamespace() == "scf" ||
-               op->getDialect()->getNamespace() == "cf") {
-      if (auto ifOp = dyn_cast<scf::IfOp>(op)) {
-        return rewriteIfOp(builder, ifOp, eraser);
-      }
+    // Rewrite `make_tensor_ptr`, `advance`, etc...
+    // Rewriting functions return the next operation to visit, or `nullptr` if
+    // there isn't one.
+    if (auto makeTensorPtrOp = dyn_cast<tt::MakeTensorPtrOp>(op))
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:678]: would something like below works?
 ```
          return TypeSwitch<Operation *, Operation *>(op)
              .Case<tt::MakeTensorPtrOp, tt::AdvanceOp, ...>([&](auto op) {
                return rewriteOp(builder, op, eraser);
              }
...
```

**Pull request merged**